src/Corax/Queries/SortingMatch.cs

```diff
@@ -292,6 +278,8 @@ public int Fill(Span<long> matches)
                 End:
                 totalMatches = kIdx;
             }
+
+            QueryContext.MatchesPool.Return(matchesKeysHolder);
```

Comment:
We intentionally avoid `try /finally`  here?

Reply:
Yes, there is actually no upside. Worst thing that can happen if you fail here --- which you shouldn't --- is increase allocations; but `try/finally` statements do disable many optimizations (afaik they are treated like `try/catch`.